### ISSUE TITLE
Full VS hang after updating to latest preview

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#40233]
i updated to:

![image](https://user-images.githubusercontent.com/4564579/70381103-a60cc700-18f9-11ea-8951-b4830de46509.png)

After a few minutes of editing i immediately hung.  Dump is available here: https://drive.google.com/file/d/1YiedU3xqZWq3dnhU4wh1C5PwTrRsAy3Z/view?usp=sharing

**Comment by CyrusNajmabadi** [comment]
Tagging @jinujoseph

**Comment by CyrusNajmabadi** [comment]
I can't get symbols working.  So no clue what's is responsible ehre.  Also, the debugger seems to be broken a *ton* of hte time with the new builds.  I can attach, but hitting 'break' often just does nothing or crashes VS.

All i can see in event-viewer (with no extra info) is:

```
Faulting application name: devenv.exe, version: 16.5.29521.150, time stamp: 0x5dd72f29
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0xddaeec2a
Faulting process id: 0x635c
Faulting application start time: 0x01d5ad498d8a07e3
Faulting application path: C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\IDE\devenv.exe
Faulting module path: unknown
Report Id: 13bc56b4-e669-496d-8998-5716d510538f
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by CyrusNajmabadi** [comment]
tagging @sharwell  @mavasani  as well.

**Comment by jinujoseph** [comment]
@CyrusNajmabadi can you file a developer community bug for the same pls

**Comment by CyrusNajmabadi** [comment]
I have tried.  But developer community doesn't seem to be responsive either.  :-/

**Comment by jmarolf** [comment]
@CyrusNajmabadi developer community issues will create a perf trace that we can look at (I believe that is why @jinujoseph is requesting this)

**Comment by CyrusNajmabadi** [comment]
Let me clarify: 

When i try to use "report a problem" i get:

![image](https://user-images.githubusercontent.com/4564579/70398469-b348b500-19d0-11ea-94e2-2dcc09f45677.png)

**Comment by jmarolf** [comment]
ah, well no much you can do then ¯\_(ツ)_/¯

**Comment by CyrusNajmabadi** [comment]
@jinujoseph @jasonmalinowski @dpoeschl  Can someone look at this.  This is a total hang, reported more than 10 days ago.  It has no traction on it at all.

**Comment by mavasani** [comment]
Main thread:

```
 	mscorlib.dll!System.Threading.SynchronizationContext.InvokeWaitMethodHelper(System.Threading.SynchronizationContext syncContext, System.IntPtr[] waitHandles, bool waitAll, int millisecondsTimeout)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	mscorlib.dll!System.Threading.Monitor.Wait(object obj, int millisecondsTimeout, bool exitContext)	Unknown
 	mscorlib.dll!System.Threading.Monitor.Wait(object obj, int millisecondsTimeout)	Unknown
 	mscorlib.dll!System.Threading.ManualResetEventSlim.Wait(int millisecondsTimeout, System.Threading.CancellationToken cancellationToken)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.SpinThenBlockingWait(int millisecondsTimeout, System.Threading.CancellationToken cancellationToken)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.InternalWait(int millisecondsTimeout, System.Threading.CancellationToken cancellationToken)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.Wait(int millisecondsTimeout, System.Threading.CancellationToken cancellationToken)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.Wait(System.TimeSpan timeout)	Unknown
 	[Waiting on Async Operation, Open Tasks View in Parallel Stacks Window to view Async Call Stacks]	
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.WaitSynchronouslyCore(System.Threading.Tasks.Task task)	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.WaitSynchronously(System.Threading.Tasks.Task task)	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTask<bool>.CompleteOnCurrentThread()	Unknown
 	Microsoft.VisualStudio.Shell.UI.Internal.dll!Microsoft.VisualStudio.Services.FileChangeService.RunWithTracing<bool>(Microsoft.VisualStudio.Services.FileChangeService.CompletionTracer tracer, System.Func<System.Threading.CancellationToken, System.Threading.Tasks.Task<bool>> func)	Unknown
 	Microsoft.VisualStudio.Shell.UI.Internal.dll!Microsoft.VisualStudio.Services.FileChangeService.RunWithTracing(Microsoft.VisualStudio.Services.FileChangeService.CompletionTracer tracer, System.Func<System.Threading.CancellationToken, System.Threading.Tasks.Task> func)	Unknown
 	Microsoft.VisualStudio.Shell.UI.Internal.dll!Microsoft.VisualStudio.Services.FileChangeService.SyncFile(string filename)	Unknown
 	Microsoft.VisualStudio.Shell.UI.Internal.dll!Microsoft.VisualStudio.Services.FileChangeService.Microsoft.VisualStudio.Shell.Interop.IVsFileChangeEx.SyncFile(string filename)	Unknown
 	Microsoft.VisualStudio.Editor.Implementation.dll!Microsoft.VisualStudio.Editor.Implementation.TextDocData.SuspendFileChangeNotification(string pszFilename, int fSuspend)	Unknown
 	Microsoft.VisualStudio.Editor.Implementation.dll!Microsoft.VisualStudio.Editor.Implementation.TextDocData.BaseSave(string pszFilename, int fRemember)	Unknown
 	Microsoft.VisualStudio.Editor.Implementation.dll!Microsoft.VisualStudio.Editor.Implementation.TextDocData.Save(string pszFilename, int fRemember, uint nFormatIndex)	Unknown
 	Microsoft.VisualStudio.Editor.Implementation.dll!Microsoft.VisualStudio.Editor.Implementation.TextDocData.Save(string pszFileName, int fRemember)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	Microsoft.VisualStudio.Editor.Implementation.dll!Microsoft.VisualStudio.Editor.Implementation.TextDocData.SaveDocData(Microsoft.VisualStudio.Shell.Interop.VSSAVEFLAGS dwSave, out string pbstrMkDocumentNew, out int pfSaveCanceled)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.SaveItem.AnonymousMethod__0(Microsoft.Internal.VisualStudio.Shell.TelemetryActivity _)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.HrMutate.AnonymousMethod__0()	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.ExecuteJob<Microsoft.VisualStudio.ProjectSystem.VS.HResult>(System.Func<System.Threading.Tasks.Task> asyncMethod, Microsoft.VisualStudio.Threading.JoinableTask job)	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.RunAsync<Microsoft.VisualStudio.ProjectSystem.VS.HResult>(System.Func<System.Threading.Tasks.Task<Microsoft.VisualStudio.ProjectSystem.VS.HResult>> asyncMethod, bool synchronouslyBlocking, Microsoft.VisualStudio.Threading.JoinableTaskCreationOptions creationOptions)	Unknown
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run<Microsoft.VisualStudio.ProjectSystem.VS.HResult>(System.Func<System.Threading.Tasks.Task<Microsoft.VisualStudio.ProjectSystem.VS.HResult>> asyncMethod, Microsoft.VisualStudio.Threading.JoinableTaskCreationOptions creationOptions)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.ProjectMultiThreadedService.ExecuteSynchronously<Microsoft.VisualStudio.ProjectSystem.VS.HResult>(System.Func<System.Threading.Tasks.Task<Microsoft.VisualStudio.ProjectSystem.VS.HResult>> asyncAction)	Unknown
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.HrInvoke.AnonymousMethod__0()	Unknown

```

Again noted extremely large number of threads with:

```
Not Flagged		20120	0	Worker Thread	Worker Thread	StreamJsonRpc.dll!StreamJsonRpc.MessageHandlerBase.Dispose
```

@CyrusNajmabadi This is likely dupe of the other deadlock issue you filed.

**Comment by CyrusNajmabadi** [comment]
Going to close out now that 16.5p2 is out.  Will open a new issue if i still run into things.
